src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/AdjournHearingAction.java

```diff
@@ -0,0 +1,183 @@
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransactionData;
+import uk.gov.hmcts.reform.sandl.snlevents.model.request.AdjournHearingRequest;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.HearingPartRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.HearingRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+import uk.gov.hmcts.reform.sandl.snlevents.service.StatusConfigService;
+import uk.gov.hmcts.reform.sandl.snlevents.service.StatusServiceManager;
+
+import javax.persistence.EntityManager;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class AdjournHearingAction extends Action implements RulesProcessable {
+
+    protected AdjournHearingRequest adjournHearingRequest;
+    protected Hearing hearing;
+    protected List<HearingPart> hearingParts;
+    protected List<Session> sessions;
+
+    protected HearingRepository hearingRepository;
+    protected HearingPartRepository hearingPartRepository;
+    protected StatusConfigService statusConfigService;
+    protected StatusServiceManager statusServiceManager;
+    protected EntityManager entityManager;
+
+    // id & hearing part string
+    private Map<UUID, String> originalHearingParts;
+    private String previousHearing;
+
+    public AdjournHearingAction(
+        AdjournHearingRequest adjournHearingRequest,
+        HearingRepository hearingRepository,
+        HearingPartRepository hearingPartRepository,
+        StatusConfigService statusConfigService,
+        StatusServiceManager statusServiceManager,
+        ObjectMapper objectMapper,
+        EntityManager entityManager
+    ) {
+        this.adjournHearingRequest = adjournHearingRequest;
+        this.hearingRepository = hearingRepository;
+        this.hearingPartRepository = hearingPartRepository;
+        this.statusConfigService = statusConfigService;
+        this.statusServiceManager = statusServiceManager;
+        this.objectMapper = objectMapper;
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public void getAndValidateEntities() {
+        hearing = hearingRepository.findOne(adjournHearingRequest.getHearingId());
+        hearingParts = hearing.getHearingParts();
+        sessions = hearingParts.stream()
+            .map(HearingPart::getSession)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        // Validation moved to act() due to conflict with optimistic lock
+
+        if (!statusServiceManager.canBeAdjourned(hearing)) {
+            throw new SnlEventsException("Hearing can not be adjourned");
+        }
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        val ids = hearing.getHearingParts().stream().map(HearingPart::getId).collect(Collectors.toList());
```

Comment:
Does it matter if you getHearingParts again? You have hearingParts already initialized in this object from getAndValidateEntities method. Could you make use of it?